### PR TITLE
Wrap powerlift incident id in Throwable on test failure

### PR DESCRIPTION
**Pull request by shahzaibj**
The output would look something like this:

java.lang.Throwable: androidx.test.uiautomator.UiObjectNotFoundException: UiSelector[RESOURCE_ID=i0118]
**PowerLift Incident Created via Intune Company Portal - 2K7S2K6Y**
at com.microsoft.identity.client.ui.automation.rules.PowerLiftIncidentRule$1.evaluate(PowerLiftIncidentRule.java:57)
at com.microsoft.identity.client.ui.automation.rules.InstallBrokerTestRule$1.evaluate(InstallBrokerTestRule.java:58)
at com.microsoft.identity.client.ui.automation.rules.BrokerSupportRule$1.evaluate(BrokerSupportRule.java:85)
at com.microsoft.identity.client.ui.automation.rules.RemoveBrokersBeforeTestRule$1.evaluate(RemoveBrokersBeforeTestRule.java:61)
at com.microsoft.identity.client.ui.automation.rules.FactoryResetChromeRule$1.evaluate(FactoryResetChromeRule.java:68)
at com.microsoft.identity.client.ui.automation.rules.CopyPreInstalledApkRule$1.evaluate(CopyPreInstalledApkRule.java:77)
at com.microsoft.identity.client.ui.automation.rules.DevicePinSetupRule$1.evaluate(DevicePinSetupRule.java:81)
at com.microsoft.identity.client.ui.automation.rules.ResetAutomaticTimeZoneTestRule$1.evaluate(ResetAutomaticTimeZoneTestRule.java:48)
at com.microsoft.identity.client.ui.automation.rules.UiAutomatorTestRule$1.evaluate(UiAutomatorTestRule.java:52)
at com.microsoft.identity.client.ui.automation.rules.RetryTestRule$1.evaluate(RetryTestRule.java:67)
at com.microsoft.identity.client.ui.automation.rules.AutomationLoggingRule$1.evaluate(AutomationLoggingRule.java:54)
at org.junit.rules.RunRules.evaluate(RunRules.java:20)
at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
at org.junit.runners.Suite.runChild(Suite.java:128)
at org.junit.runners.Suite.runChild(Suite.java:27)
at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
at androidx.test.internal.runner.TestExecutor.execute(TestExecutor.java:56)
at androidx.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:392)
at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:2145)
Caused by: java.lang.AssertionError: androidx.test.uiautomator.UiObjectNotFoundException: UiSelector[RESOURCE_ID=i0118]
at com.microsoft.identity.client.ui.automation.utils.UiAutomatorUtils.handleInput(UiAutomatorUtils.java:295)
at com.microsoft.identity.client.ui.automation.interaction.microsoftsts.AadLoginComponentHandler.handlePasswordField(AadLoginComponentHandler.java:56)
at com.microsoft.identity.client.ui.automation.interaction.microsoftsts.MicrosoftStsPromptHandler.handlePrompt(MicrosoftStsPromptHandler.java:81)
at com.microsoft.identity.client.msal.automationapp.testpass.broker.TestCase833526$1.handleUserInteraction(TestCase833526.java:88)
at com.microsoft.identity.client.msal.automationapp.sdk.MsalSdk.acquireTokenInteractive(MsalSdk.java:89)
at com.microsoft.identity.client.msal.automationapp.testpass.broker.TestCase833526.test_833526(TestCase833526.java:71)
at java.lang.reflect.Method.invoke(Native Method)
at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
at androidx.test.internal.runner.junit4.statement.RunBefores.evaluate(RunBefores.java:80)
at androidx.test.internal.runner.junit4.statement.RunAfters.evaluate(RunAfters.java:61)
at com.microsoft.identity.client.msal.automationapp.MsalLoggingRule$1.evaluate(MsalLoggingRule.java:60)
at androidx.test.rule.ActivityTestRule$ActivityStatement.evaluate(ActivityTestRule.java:531)
at com.microsoft.identity.client.ui.automation.rules.DeviceEnrollmentFailureRecoveryRule$1.evaluate(DeviceEnrollmentFailureRecoveryRule.java:50)
at com.microsoft.identity.client.ui.automation.rules.PowerLiftIncidentRule$1.evaluate(PowerLiftIncidentRule.java:33)
... 37 more
Caused by: androidx.test.uiautomator.UiObjectNotFoundException: UiSelector[RESOURCE_ID=i0118]
at androidx.test.uiautomator.UiObject.setText(UiObject.java:666)
at com.microsoft.identity.client.ui.automation.utils.UiAutomatorUtils.handleInput(UiAutomatorUtils.java:292)
... 53 more

### CODE REVIEW

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerCompanyPortal.java [hunk 1]
```diff
@@ -141,6 +141,8 @@ public void createPowerLiftIncident() {
             Assert.assertTrue(incidentIdBox.exists());
 
             Logger.w(TAG, "Incident Created with ID: " + incidentIdBox.getText());
+
+            return incidentIdBox.getText();
```
Comment (AdamBJohnsonx) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerCompanyPortal.java:145]: Maybe extract a local for this

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerMicrosoftAuthenticator.java [hunk 2]
```diff
@@ -198,21 +198,21 @@ public void enableBrowserAccess() {
...
     @Override
-    public void createPowerLiftIncident() {
+    public String createPowerLiftIncident() {
         Logger.i(TAG, "Creating Power Lift Incident..");
         launch();
         if (shouldHandleFirstRun) {
             handleFirstRun();
         }
 
         if (isInSharedDeviceMode) {
```
Comment (AdamBJohnsonx) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerMicrosoftAuthenticator.java:208]: What can change this flag at runtime?  Anything?

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/rules/PowerLiftIncidentRule.java [hunk 3]
```diff
@@ -31,21 +32,33 @@ public void evaluate() throws Throwable {
...
                     }
-                    throw originalThrowable;
+                    if (TextUtils.isEmpty(powerLiftIncidentDetails)) {
+                        throw originalThrowable;
+                    } else {
+                        final String message = originalThrowable.getMessage() + "\n" +
+                                "PowerLift Incident Created via " +
+                                ((App) powerLiftIntegratedApp).getAppName() +
+                                " - " + powerLiftIncidentDetails.trim();
+                        throw new Throwable(
```
Comment (AdamBJohnsonx) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/rules/PowerLiftIncidentRule.java:57]: Should we actually give this a type?

**Review by AdamBJohnsonx — APPROVED** [review]
Some minor comments.  Should we give the currently anonymous throwable its own type?

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerMicrosoftAuthenticator.java (hunk 2 above)
Comment (shahzaibj) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerMicrosoftAuthenticator.java:208]: Successful completion of shared device registration sets this to true here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerMicrosoftAuthenticator.java#L150

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerCompanyPortal.java (hunk 1 above)
Comment (shahzaibj) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerCompanyPortal.java]: Addressed

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/rules/PowerLiftIncidentRule.java (hunk 3 above)
Comment (shahzaibj) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/rules/PowerLiftIncidentRule.java]: Addressed here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1211/commits/62609846c5df0ead8b45fe6e2521df82af99bea5

**Pull request merged**